bluetoothServicesComponent/modem_info_service.c

```diff
@@ -0,0 +1,132 @@
+// C standard library
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+// GLib
+#include <glib.h>
+#include <gio/gio.h>
+
+// Legato
+#include "legato.h"
+#include "interfaces.h"
+
+// Local
+#include "modem_info_service.h"
+#include "org.bluez.GattCharacteristic1.h"
+#include "org.bluez.GattService1.h"
+
+#define MODEM_INFO_FSN_CHARACTERISTIC_UUID "2A25"
+#define MODEM_INFO_IMEI_CHARACTERISTIC_UUID "2A27"
+
+struct FSNContext {
+    gchar fsn[32];
+    BluezGattCharacteristic1 *bt_characteristic;
+};
+
+struct IMEIContext {
+    gchar imei[32];
+    BluezGattCharacteristic1 *bt_characteristic;
+};
```

Comment:
It seems that these context structures are not necessary.  The `bt_characteristic` members are never referenced after they are assigned.  The `fsn` and `imei` members are only used within the context of functions `handle_read_fsn_value` and `handle_read_imei_value`, so you could eliminate the context structs and just declare `gchar imei[32]` at the top of the function so that it's a stack variable.